### ISSUE TITLE
Access denied (status code 403) on static assets

### ISSUE THREAD

**Issue by MeanMangosteen** [aws-amplify/amplify-hosting#3091]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d2bialik381to3

### AWS Region

ap-southeast-2

### Amplify Hosting feature

Access control, Redirects

### Describe the bug

I'm hosting a react app on amplify with automated builds connected via github. 

My index.html requires certain javascript files, in particular this file:

`<script src="/public/datafeeds/udf/dist/bundle.js"></script>`

However once deployed, the request for this file gets a 403 error.

When make a production build locally and served locally (using same build flow as amplify), the asset is fetched fine, but once pushed to amplify it cannot fetch it.

I can see the that the `public` folder exists in the build folder on amplify as I `ls`'ed the directory after the app finished building.

These are my rules and redirects:
```
[
    {
        "source": "</^[^.]+$|\\.(?!(css|gif|ico|jpg|js|png|txt|svg|woff|woff2|ttf|map|json|webp|webm)$)([^.]+$)/>",
        "target": "/index.html",
        "status": "200",
        "condition": null
    },
    {
        "source": "/referral?id=<id>",
        "target": "/referral?id=<id>",
        "status": "301",
        "condition": null
    }
]
```

### Expected behavior

Assets are fetched successfully.

### Reproduction steps

https://add-perps.d4qy8nyhuqopw.amplifyapp.com/public/datafeeds/udf/dist/bundle.js
![image](https://user-images.githubusercontent.com/22215545/200726954-0d37d98c-78e0-4c36-a3e9-800ce8754354.png)

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by hloriana** [comment]
Hi @MeanMangosteen 👋🏽 apologies for this inconvenience. It appears that the request is being blocked by a CORS issue. We are investigating this and will provide an update shortly.

**Comment by MeanMangosteen** [comment]
Thanks for the response, your tip about it being a 404 instead was helpful. It turned out my .gitignore was misconfigured, and that file in question hadn't been pushed up. 

But I'm having another strange issue now. The response for a particular html file is incorrect.

This file:
https://add-perps.d4qy8nyhuqopw.amplifyapp.com/public/charting_library/en-tv-chart.e2a841ff.html

Is giving the same response as the index.html file:

https://add-perps.d4qy8nyhuqopw.amplifyapp.com/public/index.html

They are completely different files.

In fact when I `cat` the `en-tv-chart.e2a841ff.html` file, I can see the html stored within, but that does not come through in the response. It's just a duplicate of the index.html

The `cat` reveals (containing correct contents):
```
2022-11-15T03:14:46.545Z [INFO]: # Executing command: cat ./dist/public/charting_library/en-tv-chart.e2a841ff.html
2022-11-15T03:14:46.547Z [INFO]: <!DOCTYPE html><html lang="en" dir="ltr"><head><meta charset="utf-8"><script>window===window.parent&&(location.href="about:blank")</script><script defer="defer" crossorigin="anonymous" src="bundles/runtime.f1210af3518e525e94d7.js"></script><script defer="defer" crossorigin="anonymous" src="bundles/vendors.3197279804dd7477f3fb.js"></script><script defer="defer" crossorigin="anonymous" src="bundles/library.1fbbf36d35f97ccfbd00.js"></script><link type="text/css" href="bundles/8445.e3ed5f183ca7539d37f0.css" rel="stylesheet"></head><body class="chart-page unselectable on-widget"><div class="loading-indicator" id="loading-indicator"></div><script>var JSServer={},__initialEnabledFeaturesets=["charting_library"]</script><script>!function(){window.urlParams=function(){function t(t){return decodeURIComponent(t.replace(e," ")).replace(/<\/?[^>]+(>|$)/g,"")}for(var n,e=/\+/g,r=/([^&=]+)=?([^&]*)/g,i=function(){var t=location.href,n=t.indexOf("#");if(0<=n)return t.substring(n+1);throw"Unexpected use of this page"}(),o={};n=r.exec(i);)o[t(n[1])]=t(n[2]);var s,a=window.parent[o.uid],l=["datafeed","customFormatters","brokerFactory","save_load_adapter",
.
.
.
```

Response when served:
```
<!doctype html><html lang="en"><head><meta charset="UTF-8"/><meta content="text/html;charset=utf-8" http-equiv="Content-Type"/><meta content="utf-8" http-equiv="encoding"/><meta name="msapplication-TileColor" content="#da532c"/><meta name="theme-color" content="#141626"/><link rel="icon" type="image/svg+xml" href="/src/static/favicon/favicon.ico"/><link rel="preconnect" href="https://fonts.googleapis.com"/><link rel="preconnect" 
.
.
.
 window.EventEmitter = EventEmitter;
 window.Buffer = Buffer;</script><link rel="icon" href="favicon.ico"><script defer="defer" src="bundle.js"></script></head><body><div id="root"></div><script type="module" src="/src/main.tsx"></script></body></html>
```

As a result is causing part of the app to break.

**Comment by MeanMangosteen** [comment]
Wondering if you had a chance to look at this?

**Comment by MeanMangosteen** [comment]
I eventually figured it out. The rule mentioned above was pointing all requests for html files to the `index.html` adding the html extension to the rule above allowed the request to fall through to the correct html file.
